src/runtime/gc_conservative.go

```diff
@@ -292,9 +294,39 @@ func GC() {
 	}
 
 	// Mark phase: mark all reachable objects, recursively.
-	markGlobals()
 	markStack()
-	finishMark()
+	markGlobals()
+
+	if baremetal && hasScheduler {
+		// Channel operations in interrupts may move task pointers around while we are marking.
+		// Therefore we need to scan the runqueue seperately.
+		var markedTaskQueue task.Queue
```

Comment:
I was hoping you could just go through the runqueue instead of (essentially) rewriting it, but it seems fine. And perhaps safer.